sdk/python/testvectors/__main__.py

```diff
@@ -82,24 +134,63 @@ def create_cosignature(descriptor):
 
 	def create_aggregate(self, test_name, original_descriptor):
 		factory = self.facade.transaction_factory
-		embedded_transactions = list(map(clone_descriptor, original_descriptor['embedded']))
 
+		# create almost-ready 'printable' descriptor
+		embedded_transactions = list(map(clone_descriptor, original_descriptor['embedded']))
 		printable_descriptor = clone_descriptor(original_descriptor['aggregate'])
 		printable_descriptor['transactions'] = embedded_transactions
 		self.set_common_fields(printable_descriptor, test_name)
 		printable_descriptor['cosignatures'] = [
 			self.create_cosignature_descriptor(test_name, i) for i in range(original_descriptor.get('num_cosignatures', 0))
 		]
 
+		# fix descriptor a bit before creating transaction
 		descriptor = copy.copy(printable_descriptor)
 		descriptor['transactions'] = list(map(factory.create_embedded, embedded_transactions))
 		descriptor['transactions_hash'] = SymbolFacade.hash_embedded_transactions(descriptor['transactions'])
 		descriptor['cosignatures'] = list(map(self.create_cosignature, printable_descriptor['cosignatures']))
 
+		# fill in printable transaction hash with proper hash (not really needed/required)
 		printable_descriptor['transactions_hash'] = descriptor['transactions_hash']
 
 		return factory.create(descriptor), printable_descriptor
 
+	def create_block(self, test_name, original_descriptor):
+		block_transactions = []
+		transaction_descriptors = []
+		for entry in original_descriptor['transactions']:
+			create_transaction = self.create_aggregate if 'aggregate' in entry else self.create
+			transaction, printable_descriptor = create_transaction(test_name, entry)
+			block_transactions.append(transaction)
+			transaction_descriptors.append(printable_descriptor)
+
+		printable_descriptor = clone_descriptor(original_descriptor)
+		printable_descriptor['transactions'] = transaction_descriptors
+
+		# boring fields
+		printable_descriptor['signature'] = self.Signature(sha3.sha3_512(test_name.encode('utf8')).hexdigest())
+		printable_descriptor['signer_public_key'] = sha3.sha3_256(test_name.encode('utf8')).hexdigest()
+		printable_descriptor['timestamp'] = 0x71E71E71E71E71E0
```

Comment:
set_common_fields ?

Reply:
would have to delete deadline afterwards, so not sure if makes sense...